packages/react-scripts/per-lang-loader/PerLangPlugin.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+const dependencyTree = require('dependency-tree')
+const jsonfile = require('jsonfile')
+const path = require('path')
+const fs = require('fs')
+const rimraf = require('rimraf')
+
+const builtLocalesDir = `${__dirname}/builtLocales`
+
+class PerLangPlugin {
+  apply(compiler) {
+    const allLocales = {};
+    const manifest = {};
+    // inject a function to loaderContext so loaders can pass back info
+    compiler.hooks.beforeCompile.tapAsync('PerLangPlugin', (params, callback) => {
+      debugger;
+      if (!fs.existsSync(builtLocalesDir)) {
+        rimraf.sync(builtLocalesDir)
+        fs.mkdirSync(builtLocalesDir)
+      }
+
+      const list = dependencyTree.toList({
+        filename: `${__dirname}/src/index.js`,
+        directory: __dirname,
+        nodeModulesConfig: {
+          entry: 'module',
+        },
+        filter: (p) => p.includes('src') || p.includes('@fs'),
+      })
+
+      const realList = list.filter((p) => p.includes('locales/index.js'))
+      const allLocales = {}
+
+      realList.forEach((p) => {
+        const dir = path.dirname(p)
+        const locales = fs.readdirSync(dir).filter((d) => !d.includes('.'))
+
+        locales.forEach((locale) => {
+          const namespaces = fs.readdirSync(path.join(dir, locale)).map((d) => d.split('.')[0])
+          allLocales[locale] = allLocales[locale] || {}
+          namespaces.forEach((ns) => {
+            const strings = jsonfile.readFileSync(`${path.join(dir, locale, ns)}.json`)
+            allLocales[locale][ns] = allLocales[locale][ns] || {}
+            allLocales[locale][ns] = { ...allLocales[locale][ns], ...strings }
+          })
+        })
+      })
+
+      Object.keys(allLocales).forEach((locale) => {
+        fs.mkdirSync(path.join(builtLocalesDir, locale))
+        Object.keys(allLocales[locale]).forEach((namespace) => {
+          jsonfile.writeFileSync(`${path.join(builtLocalesDir, locale, namespace)}.json`, allLocales[locale][namespace])
+        })
+      })
+      callback()
+    });
+  }
+}
+
+module.exports = PerLangPlugin;
```

Comment:
should we commit this file if we don't use it?

Reply:
I was seeing when @tylergraf was going to change the code to the other strategy, so I left it for him